osrf/mbzirc
# Integrate comms

Requires:
  * https://github.com/ignitionrobotics/ign-gazebo/pull/1428
  * https://github.com/ignitionrobotics/ros_ign/pull/239

This pull requests integrates the comms model from Gazebo Sim. It adds the `CommsEndpoint` system to all models. The conventions used are:

* The address is the model name.
* The ROS 2 topic to send data is `<model_name>/tx`.
* The ROS 2 topic to receive data is `<model_name>/rx`.

How to test it?

1. Launch a world:
```
ros2 launch ros_ign_gazebo ign_gazebo.launch.py ign_args:="-v 4 -r simple_demo.sdf"
```

2. Spawn `quadrotor_1`:
```
ros2 launch mbzirc_ign spawn.launch.py name:=quadrotor_1 world:=simple_demo model:=mbzirc_quadrotor type:=uav x:=-0 y:=0 z:=1.2 R:=0 P:=0 Y:=0 slot3:=mbzirc_hd_camera
```

3. Spawn `quadrotor_2`:
```
ros2 launch mbzirc_ign spawn.launch.py name:=quadrotor_2 world:=simple_demo model:=mbzirc_quadrotor type:=uav x:=1 y:=1 z:=1.2 R:=0 P:=0 Y:=0 slot3:=mbzirc_hd_camera
```

4. Open a terminal and listen to the reception topic of `quadcopter_2`:
```
ros2 topic echo /quadrotor_2/rx
```

5. Open a terminal and send a message over the tx topic of `quadcopter_1`:
```
ros2 topic pub quadrotor_1/tx ros_ign_interfaces/Dataframe '{src_address: "quadrotor_1", dst_address: "quadrotor_2"}'
```

You should observe messages flowing.
